src/Lexer.php

```diff
@@ -737,27 +846,36 @@ protected function scanText()
      */
     private function getStepKeywordType(string $native): string
     {
-        // Consider "*" as a AND keyword so that it is normalized to the previous step type
-        if ($native === '*') {
-            return 'And';
+        if ($this->stepKeywordTypesCache === null) {
+            $this->stepKeywordTypesCache = [];
+            $this->addStepKeywordTypes($this->currentDialect->getGivenKeywords(), 'Given');
+            $this->addStepKeywordTypes($this->currentDialect->getWhenKeywords(), 'When');
+            $this->addStepKeywordTypes($this->currentDialect->getThenKeywords(), 'Then');
+            $this->addStepKeywordTypes($this->currentDialect->getAndKeywords(), 'And');
+            $this->addStepKeywordTypes($this->currentDialect->getButKeywords(), 'But');
         }
 
-        if (empty($this->stepKeywordTypesCache)) {
-            $this->stepKeywordTypesCache = [
-                'Given' => explode('|', $this->keywords->getGivenKeywords()),
-                'When' => explode('|', $this->keywords->getWhenKeywords()),
-                'Then' => explode('|', $this->keywords->getThenKeywords()),
-                'And' => explode('|', $this->keywords->getAndKeywords()),
-                'But' => explode('|', $this->keywords->getButKeywords()),
-            ];
+        if (!isset($this->stepKeywordTypesCache[$native])) { // should not happen when the native keyword belongs to the dialect
```

Comment:
Am I right that we can only get here if the Lexer has found a line starting with a word / phrase that is configured as a step keyword in the Dialect, and $native will be the word that it found - which must, by definition, be part of the dialect?

So I think this is a "cannot happen" rather than a "should not happen". We could probably assert, or throw a LogicException here rather than returning `Given` since this implies an internal error in the Lexer?